### ISSUE TITLE
Publish Trail Count Data to Socrata 

### ISSUE THREAD

**Issue by johnclary** [cityofaustin/atd-data-tech#8642]
Eco website > Socrata
- Create ETLs
- Create Prefect flow
- Create socrata datasets: (count data + sensor locations)

**Comment by johnclary** [comment]
We've had a long-standing backlog item to bring the trail counter data into our data ecosystem, so that staff and residents can make use of the data. Also, Rob has asked to focus on building out an executive dashboard in order to do a better job of telling the story of ATD's successes through data.

I am excited by the possibility of partnering with BP|CX, but also daunted by inter-org collaboration. I'd like to propose that we do two things.

First, if y'all can spare the time to give Charlie some iterative feedback, I think you might be surprised what Charlie can do from a data cleansing and presentation perspective using PowerBI. I'd like to get to a point where y'all are comfortable with data being used for internal analysis and also as a public resource.

Second, continue to explore how we can tap into BP|CX to benefit from the data processing work they're already doing.

**Comment by mateoclarke** [comment]
On April 19, we had a meeting Laura D, Kaz, and other DTS folks. A lot of that meeting was catching up between our respective divisions and high level priorities. They voiced interest in helping get this data more public and accessible but cited two issues that they want to resolve:
1) data validation: they want to invest some energy in manual data collection to make sure the data being collected by the devices is accurate and trustworthy
2) battery replacements: many to most of the trail counter devices had low batteries (which could also effect data quality). They've ordered replacement hardware and battery replacement installs should be happening very soon.

Charlie shared a copy of data and a visualization of some raw eco-counter data that he's been collecting through the vendor currently living on a private dataset on Socrata.

Kaz shared an attachment with a cursory analysis he did last August as an exercise to see how the free public version of Tableau was capable of showing data from for the Westgate DSDDs (and tube counters).

A follow-up meeting was scheduled for late April but ended up being cancelled.

Next steps: Mateo to reach out to Kaz to set up a new meeting including Patrick.

**Comment by Charlie-Henry** [comment]
Shelby S from TDM is interested in this data, I'm wondering if this could be an avenue to accelerate this with Laura and others?

**Comment by patrickm02L** [comment]
@Charlie-Henry will add @patrickm02L @katelunceford when meeting is scheduled.

**Comment by patrickm02L** [comment]
Does PR [#76](https://github.com/cityofaustin/atd-prefect/pull/76) close this Issue?

**Comment by johnclary** [comment]
@patrickm02L this is currently only available internally. it's a flip of a switch to make it public but iirc we got resistance from department staff. we should open it up and ask for forgiveness.

<img width="1204" alt="Screenshot 2023-05-05 at 11 56 48 AM" src="https://user-images.githubusercontent.com/14793120/236508013-d751c4ed-926d-432c-9a31-42789ba04770.png">

**Comment by johnclary** [comment]
and fwiw this data _is_ publicly available by scraping the vendor website—which in IMO is a reason why we should make this dataset. public

**Comment by Charlie-Henry** [comment]
@johnclary I was poking around the API for the eco-counters and I found the way to get it by hourly now 

Southern Walnut Creek Trail Counter metadata, note `token`:
https://www.eco-visio.net/api/aladdin/1.0.0/pbl/publicwebpage/100020993?withNull=true

Southern Walnut Creek Trail count (comptage 🇫🇷 ) by hour: 
https://www.eco-visio.net/api/aladdin/1.0.0/pbl/publicwebpage/data/100020993?begin=20221001&end=20221027&step=3&domain=89&withNull=true&t=4eb055ec965c6627995dc5c19d4767b80c4457f2e5110f00e737ddc07cfb50e8

`t=` is the token from the metadata page.

This [repo](https://github.com/bundesAPI/eco-visio-api) had some good examples.

**Comment by Charlie-Henry** [comment]
I have opened a PR for the ETL that is now supporting two datasets in Socrata:
- [24 Hour Daily Totals](https://datahub.austintexas.gov/d/26tt-cp67)
- [Device metadata](https://datahub.austintexas.gov/d/vxcr-pjs7)

I would like to be able to publish the hourly totals but I ran into some issues with permissions. Apparently only two of our devices actually allow me to get the hourly data. Maybe something with could change in the future?
